runtime/fundamentals/modules.md

```diff
@@ -0,0 +1,450 @@
+they provide URL access to JavaScript modules:
+
+- [deno.land/x](https://deno.land/x)
+- [esm.sh](https://esm.sh)
+- [unpkg.com](https://unpkg.com)
+
+URL imports are useful if you have a small, often single file, Deno project that
+doesn't require any other configuration because you can avoid having a
+`Deno.json` file at all. It's not advised to use this style of import in larger
+applications as you may end up with version conflicts (where different files use
+different version specifiers).
+
+From a security perspective, the contents of files at URLs can change, so we do
+not generally recommend this approach for third-party components.
+
+URL imports remain supported, **but we recommend using a package registry for
+the best experience.**
+
+### Overriding URL imports
+
+The other situation where import maps can be very useful is to override URL
+imports in specific modules.
+
+Let's say you want to override a `https://deno.land/x/my-library@1.0.0/mod.ts`
+specifier that is used inside files coming from `https://deno.land/x/example/`
+to a local patched version. You can do this by using a scope in the import map
+that looks something like this:
+
+```json
+{
+  "imports": {
+    "example/": "https://deno.land/x/example/"
+  },
+  "scopes": {
+    "https://deno.land/x/example/": {
+      "https://deno.land/x/my-library@1.0.0/mod.ts": "./patched/mod.ts"
+    }
+  }
+}
+```
+
+:::note
+
+URL imports have no notion of packages. Only the import map at the root of your
+project is used. Import maps used inside URL dependencies are ignored.
+
+:::
+
+## Proxies
+
+Deno is able to handle network requests through a proxy server, useful for
+various reasons such as security, caching, or accessing resources behind a
+firewall. The runtime supports supports proxies for module downloads and the Web
+standard `fetch` API.
+
+Deno reads proxy configuration from environment variables: `HTTP_PROXY`,
+`HTTPS_PROXY` and `NO_PROXY`.
+
+In case of Windows, if environment variables are not found, Deno falls back to
+reading proxies from the registry.
+
+## Publishing modules
+
+Any Deno program that defines an export can be published as a module. This
+allows other developers to import and use your code in their own projects.
+Modules can be published to [JSR](https://jsr.io), the modern JavaScript and
+TypeScript registry. Check out the
+[JSR documentation on publishing modules](https://jsr.io/docs/publishing-packages)
+for more information.
+
+## Publishing for Node.js
+
+You can make your Deno modules available to Node.js users with the
+[dnt](https://github.com/denoland/dnt) build tool.
+
+dnt allows you to develop your Deno module mostly as-is and use a single Deno
+script to build, type check, and test an npm package in an output directory.
+Once built, you only need to `npm publish` the output directory to distribute it
+to Node.js users.
+
+For more details, see
+[https://github.com/denoland/dnt](https://github.com/denoland/dnt).
```

Comment:
Modules can be published either to JSR or NPM or accessed via HTTPS (deno.land/x being one option). This needs to be reworked.